yarn-project/p2p/src/testbench/port_change.test.ts

```diff
@@ -3,43 +3,56 @@ import { sleep } from '@aztec/foundation/sleep';
 import { ClientIvcProof } from '@aztec/stdlib/proofs';
 import { mockTx } from '@aztec/stdlib/testing';
 
-import { assert } from 'console';
 import getPort from 'get-port';
 import path from 'path';
 import { fileURLToPath } from 'url';
 
+import type { P2PConfig } from '../config.js';
 import { WorkerClientManager, testChainConfig } from './worker_client_manager.js';
 
-const logger = createLogger('testbench-ports');
 const NUMBER_OF_ITERATIONS = 2;
-const NODES_TO_CHANGE_PORT = 5;
+const NODES_TO_CHANGE_PORT = 1;
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
 // Test Summary:
-// - Start 20 clients
+// - Start 5 clients
 // - For NUMBER_OF_ITERATIONS iterations:
 //    - Send a tx from a random client
 //    - Allow for it to propagate to all other clients
 //    - change the port for NODES_TO_CHANGE_PORT random clients
 //    - Wait for two peer manager heartbeats
 describe('Port Change', () => {
+  let workerClientManager: WorkerClientManager;
+  let numberOfClients: number;
+  let testConfig: P2PConfig;
+  const logger = createLogger('testbench-ports');
+
+  beforeEach(async () => {
+    logger.info('Starting test setup');
+    // Use 5 node configuration for this test
+    const configPath = path.join(__dirname, '../../testbench/configurations', 'normal-degree-5-nodes.json');
+    logger.info(`Loading config from ${configPath}`);
+    const config = await import(configPath, { assert: { type: 'json' } });
+    testConfig = {
+      ...testChainConfig,
+      ...config.default,
+      boostrapNodeEnrVersionCheck: false,
+      bootstrapNodesAsFullPeers: true,
+    };
+    numberOfClients = config.default.numberOfClients;
+    logger.info(`Creating ${numberOfClients} clients`);
+
+    workerClientManager = new WorkerClientManager(logger, testConfig);
+    await workerClientManager.makeWorkerClients(numberOfClients);
+
+    // wait a bit longer for all peers to be ready
+    await sleep(10000);
+    logger.info('Workers Ready');
+  });
```

Comment:
Heads up that hooks have a shorter timeout than the test. Not sure how expensive is this setup is (or the cleanup in the new `afterEach` hook), but you may want to bump the timeout for them if needed.